Lesson3/Program.cs

```diff
@@ -28,17 +28,15 @@ public static List<string> GetSortedData(string[] fileData)
         [STAThread]
         public static void Main(string[] clArguments)
         {
-            if (clArguments.Any(a => String.Compare(a, "-json", true) == 0) || clArguments.Any(a => String.Compare(a, "-xml", true) == 0))
+            if (clArguments.Any(a => String.Compare(a, "-xml", true) == 0))
             {
-                //use interface
-                IFileHandling ifhJson = GetJsonParser();
-                IFileHandling ifhXml = GetXmlParser();
+                IFileHandling ifhXml = new XmlParser();
```

Comment:
Observation: Per our discussion, this should have just replaced the null that should have been there.